src/lib/one-graph/cli-client.js

```diff
@@ -531,26 +726,48 @@ const getCLISession = async ({ jwt, oneGraphSessionId, siteId }) => {
  * @param {string} input.siteId The site object that contains the root file path for the site
  */
 const getCLISessionMetadata = async ({ jwt, oneGraphSessionId, siteId }) => {
-  const { errors, session } = await getCLISession({ jwt, oneGraphSessionId, siteId })
+  const result = await getCLISession({ jwt, oneGraphSessionId, siteId })
+  if (!result) {
+    warn(`Unable to fetch CLI session metadata`)
+  }
+  const { errors, session } = result
   return { metadata: session && session.metadata, errors }
 }
 
 /**
  * Look at the current project, filesystem, etc. and determine relevant metadata for a cli session
  * @param {object} input
  * @param {string} input.siteRoot The root file path for the site
- * @returns {Record<string, any>} Any locally detected facts that are relevant to include in the cli session metadata
+ * @param {object} input.config The parsed netlify.toml config file
+ * @returns {Promise<CliEventHelper.DetectedLocalCLISessionMetadata>} Any locally detected facts that are relevant to include in the cli session metadata
  */
-const detectLocalCLISessionMetadata = ({ siteRoot }) => {
+const detectLocalCLISessionMetadata = async ({ config, siteRoot }) => {
   const { branch } = gitRepoInfo()
   const hostname = os.hostname()
   const userInfo = os.userInfo({ encoding: 'utf-8' })
   const { username } = userInfo
   const cliVersion = USER_AGENT
 
-  const config = loadNetlifyGraphConfig(siteRoot)
+  const editor = process.env.EDITOR || null
```

Comment:
wanna do a `warn()` here to instruct users to set `$EDITOR`?

Reply:
As a nice-to-have: We should probably should have a specific 'checkCliEnvOnStartupAndMaybeWarnUser' that checks everything and emits warnings once on startup. Punting on it for now.

Reply:
(The UI will also have reminders to set `$EDITOR` if it detects none is configured after a file is written to disk)